ui-components/src/helpers/unitTypes.ts

```diff
@@ -21,7 +21,7 @@ export const getUniqueUnitTypes = (units: Unit[]): GetUnitTypeNamesReturn[] => {
 
     if (!id || !name) return acc
 
-    const unitTypeExists = acc.find((item) => item.id === id)
+    const unitTypeExists = acc.some((item) => item.id === id)
```

Comment:
I think the main difference here is that `unitTypeExists` is used as a boolean flag so using `.some` here gives us a boolean while `.find` gives us back the "item" if that makes sense

Its definitely not a big difference just something I saw while I was going through this file

If there is no chance for duplicates then, the difference is `.some` returns false while `.find` gives us undefined. Although if there is no chance for duplicates it might be better to just remove this code entirely since it is requiring a loop inside of a loop